packages/svelte/src/internal/client/runtime.js

```diff
@@ -1156,6 +1162,11 @@ export function managed_pre_effect(init, sync) {
  * @returns {import('./types.js').EffectSignal}
  */
 export function pre_effect(init) {
+	if (current_effect === null) {
+		throw new Error(
+			'The Svelte $effect.pre rune can only be used during component initialisation.'
```

Comment:
is this accurate? [this sort of thing works](https://svelte-5-preview.vercel.app/#H4sIAAAAAAAAE22RzW6DMAzHXyWKegCtauk1A6Qe9wxjBxZMFy04KHGQKpR3X_goVOrkS_z1s_P3yFulwXHxOXKsO-CCX_ueHznd-8lxA2iC6DvjrZwiuZNW9VRWWJEGYtJ4JFawg6OaIMnS95iJudajJGWQKZQWOkBKUjZOmYqWnreCXaZiCkuHNOgmHpJVAzSRuZbf4hj6UXjbCRUdoG1B0qm3kMR4Ue6pB0ohgR1qHUkO6GP1XqtfFlosHNkly7YPLWaBvEX2H0NDbbchj9npM3B7h2foipx3WCvCKkt-3tXG_NsTRUENCqmV_C3GTdown2OOOsHGTcPTLNsMWprLeMrONKpV0HBB1kP4Cn9tt770BQIAAA==), and we've prototyped using this sort of technique for store-like behaviour.

Then again obviously the vast majority of cases _are_ component initialization. I'm not sure what language would be best to use here

Reply:
Yeah, that case will leak if there's no parent effect unfortunately. It slipped through.